ui/components/src/model/legend.ts

```diff
@@ -24,15 +24,23 @@ export interface LegendOptions {
 export interface LegendItem {
   id: string;
   label: string;
-  isSelected: boolean;
   color: string;
-  onClick: MouseEventHandler<HTMLElement>;
+  onClick?: MouseEventHandler<HTMLElement>;
 }
 
 export type LegendPositionConfig = {
   label: string;
 };
 
+/**
+ * State of selected items in the legend.
+ * - When "ALL", all legend items are selected, but not visually highlighted.
+ * - Otherwise, it is a Record that associates legend item ids with a boolean
+ *   value. When the associated entry for a legend item is `true`, that item
+ *   will be treated as selected and visually highlighted.
+ */
+export type SelectedLegendItemState = Record<LegendItem['id'], boolean> | 'ALL';
```

Comment:
I wonder if we need `ALL` here because i think it should be up to the consumers to decide what it means when none is selected

Reply:
I think it's reasonable for the Legend to be a little opinionated about the user experience/interaction for consistency across use and enabling some reasonable defaults for the behavior. The "ALL" is basically a special handling case for the behavior of "everything is selected, but not visually" that we want by default in the list behavior. 

The `selected` and `onSelectedItemsChange` props make this a controlled component, so the consumer can adjust the values if they want some slightly more custom behavior. If the consumer wants a true "none selected," they can set `selected` to an empty record or a record with all the entries set to false. If they want a "select all and visually select all," they can set `selected` to a record with all the entries set to true. 

Similar to the comment above, I'm hoping this special "ALL" case can eventually go away if we swap the list legend to checkboxes, but I don't want to do that in this particular PR.